src/maintainer/knowledge_base.rst

```diff
@@ -1136,3 +1136,112 @@ key with ``docker_image``. Also ``cdt_name`` ensures the CDTs match the CentOS
 
 Finally, note that the ``aarch64`` and ``ppc64le`` platforms already use CentOS 7.
+
+.. _cuda:
+
+CUDA builds
+===========
+
+Although the provisioned CI machines do not feature a GPU, Conda Forge does provide mechanisms
+to build CUDA-enabled packages. These mechanisms involve several packages:
+
+* ``cudatoolkit``: The runtime libraries for the CUDA toolkit. This is what end-users will end
+  up installing next to your package.
+
+* ``nvcc``: Nvidia's EULA does not allow the redistribution of compilers and drivers. Instead, we
+  provide a wrapper package that locates the CUDA installation in the system. The main role of this
+  package is to set some environment variables (``CUDA_HOME``, as well as ``CFLAGS`` and friends),
+  as well as wrapping the real ``nvcc`` executable to set some extra command line arguments.
+
+In practice, to enable CUDA on your package, add ``{{ compiler('cuda') }}`` to the ``build``
+section of your requirements and rerender! The matching ``cudatoolkit`` will be added to the ``run``
+requirements automatically.
+
+.. note::
+
+  **How is CUDA provided at the system level?**
+
+  * On Linux, Nvidia provides official Docker images, which we then
+    `adapt <https://github.com/conda-forge/docker-images>`_ to the Conda Forge needs.
+
+  * On Windows, the compilers need to be installed for every CI run. This is done through the
+    `conda-forge-ci-setup <https://github.com/conda-forge/conda-forge-ci-setup-feedstock/>`_ scripts.
+    Do note that the Nvidia executable won't install the drivers because no GPU is present in the machine.
+
+  **How is ``cudatoolkit`` selected at install time?**
+
+  Conda exposes the maximum CUDA version supported by the installed Nvidia drivers through a virtual package
+  named ``__cuda``. By default, ``conda`` will install the highest version available
+  for the packages involved. However, prior to v4.8.4, ``__cuda`` versions would not be part of the
+  constraints, so you would always get the latest one, regardless the supported CUDA version.
+
+  If for some reason you want to install a specific version, you can use::
+
+    conda install your-gpu-package cudatoolkit==10.1
+
+Testing the packages
+--------------------
+
+Since the CI machines do not feature a GPU, you won't be able to test the built packages as part
+of the conda recipe. For now, the workaround is to enable the Azure artifacts for your feedstock
+(see :ref:`azure-config`) and then perform the tests locally.
+
+Common problems and known issues
+--------------------------------
+
+Helping your build system locate CUDA
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Some build systems might need some help finding CUDA, so you might need to pass the value
+of ``CUDA_HOME`` as some kind of flag. Some examples include:
+
+* Old-style CMake (using the deprecated ``FindCUDA`` module), you might need to pass this flag::
+
+    -DCUDA_TOOLKIT_ROOT_DIR=${CUDA_HOME}
```

Comment:
This should be added to `CMAKE_ARGS` env variable and users should be encouraged to use `${CMAKE_ARGS}`

Reply:
Not sure as this is a legacy thing. The `CUDA_HOME` is already handled by the `nvcc` wrapper. As libraries move away from `FindCUDA`, the shouldn't need the `CUDA_TOOLKIT_ROOT_DIR` argument either. Maybe it's ok to just document and eventually remove from the docs (or move to a legacy section)?

Reply:
Once again, this _shouldn't_ be needed since `CUDA_PATH` is defined properly by the `nvcc` wrapper and `FindCUDA` [will look](https://gitlab.kitware.com/cmake/cmake/-/blob/v3.19.2/Modules/FindCUDA.cmake#L689) there for `nvcc`, but since I haven't checked all the possible scenarios, I am leaving that here (hence the _might_) in case it helps somebody running into a corner case.

Reply:
Ah wait, `nvcc` only defines `CUDA_PATH` on Windows. We might want that on Linux too.

Reply:
`FindCUDAToolkit` also relies on `CUDA_PATH`: https://gitlab.kitware.com/cmake/cmake/-/blob/v3.19.2/Modules/FindCUDAToolkit.cmake#L32-33

Reply:
Should be fixed in `nvcc-feedstock` instead of asking the user to do this in recipes.

Reply:
I'll open a PR later today / tomorrow.

Reply:
PR here: https://github.com/conda-forge/nvcc-feedstock/pull/58